libcontainer/intelrdt/intelrdt.go

```diff
@@ -152,9 +152,13 @@ type Manager struct {
 	path   string
 }
 
-// NewManager returns a new instance of Manager, or nil, if the Intel RDT
-// functionality is not available from hardware or not enabled in the kernel.
+// NewManager returns a new instance of Manager, or nil if the Intel RDT
+// functionality is not specified in the config, available from hardware or
+// enabled in the kernel.
 func NewManager(config *configs.Config, id string, path string) *Manager {
+	if config.IntelRdt == nil {
+		return nil
+	}
```

Comment:
This means that the further checks for `m.config.IntelRdt != nil` can be dropped.

Can be addressed as a follow up for sure.

Reply:
Is that safe to do? `m.config` is a pointer, so `m.config.IntelRdt` could be potentially set to nil after the manager is constructed. Right now, with all dereferences of `m.config` guarded, doing so would not result in a nil-dereference when calling any manager methods. I worry removing the guards could be a breaking change for some non-runc consumers of `libcontainer/intelrdt`.

Reply:
Makes sense; since the config we set is not a copy but merely a pointer, it can be changed. Let's leave these as is.